cupy/cuda/stream.pyx

```diff
@@ -177,10 +177,11 @@ class Stream(object):
             self.ptr = runtime.streamCreate()
 
     def __del__(self):
+        cdef void* current_ptr
         if self.ptr:
             tls = _StreamThreadLocal.get()
             current_ptr = tls.get_current_stream_ptr()
-            if <size_t>self.ptr == <size_t>current_ptr:
+            if <void*>self.ptr == current_ptr:
```

Comment:
I think it is this line causing the segfaults in CI when the stream object is deleted or out of scope. The comparison never evaluates to `True` because `<void*>self.ptr` is (I think) the pointer to the Python object `self.ptr`, not the pointer to the underlying stream, as can be checked by casting back to an address: `<intptr_t><void*>self.ptr != current_ptr`.

Reply:
Fixed in e9a655b. @niboshi could you kick off Jenkins one more time for me please?

Reply:
Sorry, my suggestion was wrong.
As you pointed out, In Cython `<void*>` cast is semantically totally different from cast to integer types...
Another way around would be to double cast to `void*` like `<void*><intptr_t>self.ptr`, but I think current fix  e9a655bbf3228da72fed47a0d40b14ac6afcccc6 is also acceptable.

Reply:
Cython can be surprising from time to time...